### ISSUE TITLE
fake updates and other weirdness after releasing older image

### ISSUE THREAD

**Issue by rade** [weaveworks/flux#709]
We recently encountered a bug in one of our services, with version A of one of its two images, and re-released an earlier version B. Now whenever an image for another service is released, the [git commit](https://github.com/weaveworks/service-conf/commit/e599eca139953b56965a8feb1d8e208fe8532d3f) additionally mentions an update from B to A
```
Release quay.io/weaveworks/logging:master-ae8afaa, quay.io/weaveworks/ui-server:master-6024fa2 to automated
```
even though the contents of the commit make no such change, i.e. in this example actually only the ui-server image is updated.

The slack notification is wrong too:
![screenshot from 2017-08-19 07-48-09](https://user-images.githubusercontent.com/109109/29484279-1530dbc2-84b3-11e7-86de-b02f02a76dc2.png)

A related issue is that it is not clear *why* flux is not changing the image to the latest version again - a good thing in this case, but still mysterious. The UI for the service looks like this:
![screenshot from 2017-08-19 07-52-12](https://user-images.githubusercontent.com/109109/29484304-897d57f8-84b3-11e7-9908-ea5a45622b2f.png)
Note that the service is automated and unlocked, and has no tag filters. To me that means flux should always be releasing the latest version. Yet the "Update" button is greyed out, and the logging image is clearly shown as not being at the latest version.

**Comment by squaremo** [comment]
I suspect that it's calculating that it needs to update those images, then failing to apply the change to logging. But because it succeeds at changing _something_, there's a diff to commit, so it proceeds.

The surprising bit is that it silently fails to make the update to the logging image. It should either make the change, or bail out.

**Comment by squaremo** [comment]
Fixed by #720
